tests/DoctrineObjectTest.php

```diff
@@ -2978,4 +2979,49 @@ public function testWrongEnumBackedValueThrowsException(): void
         $this->hydratorByValue->addStrategy('enum', new Assets\SimpleEnumStrategyPhp81());
         $this->hydratorByValue->hydrate($data, $entity);
     }
+
+    /**
+     * @requires PHP 8.1
+     */
+    public function testExtractReadonlyProps(): void
+    {
+        $entity = new Assets\SimpleEntityWithReadonlyPropsPhp81(2);
+
+        $this->configureObjectManagerForSimpleEntity(Assets\SimpleEntityWithReadonlyPropsPhp81::class);
+
+        $data = $this->hydratorByReference->extract($entity);
+        $this->assertEquals([], $data);
+
+        $entity->setField('value');
+        $data = $this->hydratorByReference->extract($entity);
+        $this->assertEquals(['field' => 'value'], $data);
+    }
+
+    /**
+     * @requires PHP 8.1
+     */
+    public function testHydrateReadonlyProps(): void
+    {
+        $this->configureObjectManagerForSimpleEntity(Assets\SimpleEntityWithReadonlyPropsPhp81::class);
+
+        $entity = $this->hydratorByValue->hydrate(['field' => 'toto', 'id' => 15], new Assets\SimpleEntityWithReadonlyPropsPhp81(3));
+
+        $this->assertEquals(3, $entity->getId());
+        $this->assertEquals('toto', $entity->getField());
+    }
+
+    /**
+     * @requires PHP 8.2
+     */
+    public function testExtractReadonlyClass(): void
```

Comment:
Could you please add a test `testHydrateReadonlyClass()` as well?

For this, you will need to add a check and throw of `LogicException`  in `DoctrineObject::hydrateByReference()`.

Reply:
done